astropy/coordinates/baseframe.py

```diff
@@ -1018,7 +1096,19 @@ def spherical(self):
 
         # TODO: if representations are updated to use a full transform graph,
         #       the representation aliases should not be hard-coded like this
-        return self.represent_as(SphericalRepresentation, in_frame_units=True)
+        return self.represent_as('spherical', in_frame_units=True)
+
+    @property
+    def sphericalcoslat(self):
+        """
+        Shorthand for a spherical representation of the coordinates in this object.
```

Comment:
Copy-pasted from spherical, fixed that as well. 

Also fixed a bug I found in `sphericalcoslat` and will add a test in the next PR, but I realized it won't actually work in this PR because of some code that needs to be in the frame initializer...